### DISCUSSION TITLE
About this aidl and xp framework to communicate with the service is now available?

### DISCUSSION THREAD

**Discussion by etjson** [LSPosed/LSPosed#1791] [Q&A] [answered]
[https://github.com/libxposed/XposedService/issues/1](https://github.com/libxposed/XposedService/issues/1)

About this aidl and xp framework to communicate with the service is now available?

Need to use setModuleScope to easily help the user hook up the application.

If it's available now, is there a full demo available for reference.

**Comment by mywalkb** [comment, marked as answer]
No is not available and maybe will never be!
The lsposed team boycott these types of access to manager.
No cli, no module api, how you can see in issue they are very old and if someone try to develop them they don't accept PR.
So it's great LSPosed for scope but it's bad because only manager can modify them.
If you try to change with direct access to database with stop and start, now when lspd start reboot the device.

**Reply by NULL0B** [reply]
This is sad, it was something I was looking forward to.

**Reply by mywalkb** [reply]
> This is sad, it was something I was looking forward to.

I forked the repository in mywalkb/LSPosed_mod, in my version CLI is implemented, the patch is not mine is taken from a PR not merged.
In todo list there is also module api. I think is more important module api and cli than gui and colors.
I will keep them in sync as much as possible. I'm not expert as members of LSPosed team but I will do my best.
